File: backend/api/rankings.py
from fastapi import APIRouter, Query
from backend.sensor.sensor_controller import SensorController


def create_rankings_router(sensor_controller: SensorController) -> APIRouter:
    router = APIRouter()

    @router.get("/")
    def list_rankings(
        sort_by: str = Query("aqi", alias="sortBy", regex="^(aqi|temp|noise|humidity)$"),
        search: str = Query("", alias="search"),
    ):
        return sensor_controller.get_rankings(sort_by=sort_by, search=search)

    return router

File: frontend/src/lib/api/real/rankings.ts
import type { City } from "../types";
import { request } from "./client";

export type RankingSort = "aqi" | "temp" | "noise" | "humidity";

export async function listRankings(
  sort: RankingSort = "aqi",
  search = ""
): Promise<City[]> {
  const query = new URLSearchParams({ sortBy: sort, search });
  return request<City[]>(`/api/v1/rankings?${query.toString()}`);
}
